src/main/java/redis/clients/jedis/JedisClusterInfoCache.java

```diff
@@ -33,19 +33,32 @@
   private String password;
   private String clientName;
 
+  private boolean ssl;
+  private SSLSocketFactory sslSocketFactory;
+  private SSLParameters sslParameters;
+  private HostnameVerifier hostnameVerifier;
+  private JedisClusterPortMap portMap;
+
   private static final int MASTER_NODE_INDEX = 2;
 
   public JedisClusterInfoCache(final GenericObjectPoolConfig poolConfig, int timeout) {
-    this(poolConfig, timeout, timeout, null, null);
+    this(poolConfig, timeout, timeout, null, null, false, null, null, null, null);
   }
 
   public JedisClusterInfoCache(final GenericObjectPoolConfig poolConfig,
-      final int connectionTimeout, final int soTimeout, final String password, final String clientName) {
```

Comment:
This breaks backward compatibility. Keep this constructor and add another one. It is possible to call the new constructor from here.